test/lit/ctor-eval/partial-global.wat

```diff
@@ -0,0 +1,31 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: wasm-ctor-eval %s --ctors=test --kept-exports=test --quiet -all -S -o - | filecheck %s
+
+(module
+ ;; CHECK:      (type $none_=>_none (func))
+
+ ;; CHECK:      (global $global (mut i32) (i32.const 0))
+ (global $global (mut i32) (i32.const 0))
+
+ (func "test"
+  ;; The nop can be evalled away, but not the loop. We should not apply any
+  ;; partial results from the loop - in particular, the global must remain at
+  ;; 0.
+  (nop)
+  (loop
+   (global.set $global
+    (i32.const 999) ;; this must not be applied to the global!
```

Comment:
Yes, I guess in this simple module it isn't observable. But in general that would be dangerous. I'm adding this comment now:
```wat
  ;; (It is true that in this simple module it would be ok to set 999 to the
  ;; global, but if the global were exported for example then that would not
  ;; be the case, nor would it be the case if the code did $global = $global + 1
  ;; or such. That is, since the global.set is not evalled away, its effects
  ;; must not be applied; we do both atomically or neither, so that the
  ;; global.set's execution only happens once.)
```